roles/db-copy/tasks/active-copy.yml

```diff
@@ -0,0 +1,155 @@
+  changed_when: false
+  register: pmon_proc
+  tags: active-duplicate
+
+- name: Active-copy | Add oratab entry
+  lineinfile:
+    path: /etc/oratab
+    regexp: '^{{ oracle_sid }}\:'
+    line: '{{ oracle_sid }}:{{ oracle_home }}:N'
+    owner: "{{ oracle_user }}"
+    group: "{{ oracle_group }}"
+  become: yes
+  tags: active-duplicate
+
+- name: Active-copy | Set random password
+  set_fact:
+    sys_pass: "{{ lookup('password', '/dev/null length=16 chars=ascii_letters,digits') }}0#_"
+  tags: active-duplicate
+
+- name: Active-copy | Set sys password for primary db
+  shell: |
+    set -o pipefail
+    {{ oracle_home }}/bin/orapwd file={{ oracle_home }}/dbs/orapw{{ db_name }} force=y password={{ sys_pass }}
+  environment:
+    ORACLE_HOME: "{{ oracle_home }}"
+    ORACLE_SID: "{{ oracle_sid }}"
+    PATH: "{{ oracle_home }}/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin"
+  delegate_to: primary1
+  become: yes
+  become_user: "{{ oracle_user }}"
+  tags: active-duplicate
+
+- name: Active-copy | Add static listener entry
+  lineinfile:
+    path: "{{ grid_home }}/network/admin/listener.ora"
+    regexp: "^SID_LIST_{{ listener_name }}"
+    line: "SID_LIST_{{ listener_name }}=(SID_LIST=(SID_DESC=(GLOBAL_DBNAME={{ oracle_sid }}{{ standby_suffix }}.{{ db_domain }})(ORACLE_HOME={{ oracle_home }})(SID_NAME={{ oracle_sid }})))"
+    owner: "{{ grid_user }}"
+    group: "{{ oracle_group }}"
+  become: yes
+  tags: active-duplicate
+
+- name: Active-copy | Reload listener
+  shell: "{{ grid_home }}/bin/lsnrctl reload {{ listener_name }}"
+  environment:
+    ORACLE_HOME: "{{ grid_home }}"
+    PATH: "{{ grid_home }}/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin"
+  become: yes
+  become_user: "{{ grid_user }}"
+  tags: active-duplicate
+
+- name: Active-copy | Create directories
+  file:
+    path: "{{ oracle_base }}/admin/{{ oracle_sid }}/adump"
+    state: directory
+    owner: "{{ oracle_user }}"
+    group: "{{ oracle_group }}"
+  become: yes
+  tags: active-duplicate
+
+- name: Active-copy | Create auxiliary init file
+  shell: |
+    set -o pipefail
+    (echo "db_name={{ db_name }}"; echo "db_unique_name={{ db_name }}{{ standby_suffix }}") > {{ oracle_home }}/dbs/init{{ oracle_sid }}.ora
+  become: yes
+  become_user: "{{ oracle_user }}"
+  tags: active-duplicate
+
+- name: Active-copy | Start auxiliary instance
+  shell: |
+    set -o pipefail
+    {{ oracle_home }}/bin/sqlplus -s / as sysdba << EOF
+    startup nomount pfile={{ oracle_home }}/dbs/init{{ oracle_sid }}.ora force
+    alter system register;
+    host sleep 60
+    EOF
+  environment:
+    ORACLE_HOME: "{{ oracle_home }}"
+    ORACLE_SID: "{{ oracle_sid }}"
+    PATH: "{{ oracle_home }}/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin"
+  become: yes
+  become_user: "{{ oracle_user }}"
+  tags: active-duplicate
+
+- name: Active-copy | Set sys password for auxiliary instance
```

Comment:
What do you think about backing up the previous password file?  Just in case one was created manually and may not remember the password.

Reply:
I do not think it is necessary to have backup of auxiliary instance password, but for primary it might have sense. However thinking about it, the password file can already have other users assigned and such brute overwrite will erase them from password file. Do we need to check and take care of others assigned in password file?

Reply:
Even if there's just a SYSDBA password, we do overwrite it.  So maybe a good idea to save it somewhere?  (And maybe somehow take care of the case when two duplicates are run in a row and not to clobber the backup)

Reply:
I added tasks to backup password files with pattern up to minutes and will put it together with rman block changes